src/main/java/org/fcrepo/storage/ocfl/OcflObjectSession.java

```diff
@@ -56,11 +57,20 @@
 
     /**
      * Deletes a resource's header file. This should only be called if its content file has already been deleted.
+     * If the resource is the root resource of the object, then the object will be deleted.
```

Comment:
Not that I disagree with this change, just curious what prompted it? What circumstances are we anticipating deleting the header file but not the content?

Reply:
Currently, the code operates under the assumption that if you're calling `deleteHeaderFile` then you already called `deleteContentFile`. At the cost of an additional header read, it could enforce that the content file has already been deleted and reject `deleteHeaderFile` requests when the content file still exists. But, if we did that, there's no real benefit to having `deleteHeaderFile` and we may as well get rid of it and just have `deleteContentFile` and `deleteResource`, which I'm certainly open to.

Reply:
I removed the `deleteHeaderFile` api